keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -1225,6 +1226,97 @@ Note that given the estimation for duration of processing the requests is
 automatically corrected (both up and down), there is no need to change that
 in the initial version.
 
+
+#### Estimate width of LIST request
+
+Terms:
+- `W`: estimated width of the request
+- `N`: number of object(s) estimated to be processed for serving this `LIST` request
+- `L`: value of the user specified `limit` in the request URI
+- `S`: an integer value that represents the semantics of the `LIST` request, we use as a multiplying 
+   factor when we estimate the width.
+
+
+What's not in scope for this iteration:
+- `namespaced` or `cluster scoped`: we will not take this into account while estimating
+  the width of a `LIST`
+- We are not going to use to the Priority Level configuration assigned to this request for width estimation.   
+- Use the watch cache interface to know whether this `LIST` will be served from the cache.
+- The code site (inside p&f server filter) where we inspect the request query parameters for `LIST`
+  is well before these parameters are validated. We will not validate these parameters when we estimate
+  the width. We will ignore the fact that the user might have provided invalid data semantically.
+- We are going to treat `resourceVersion` as opaque, its value will not be used to estimate `N`.   
+
+While determining the width, we keep in sight the following goals:
+- We will inspect the request query parameters and estimate width for the `LIST` request.
+- Be pessimistic, assume the worst. We will possibly overestimate the `width` for certain LIST request.
+
+
+Estimate `S`:
+
+These are the semantics (from https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions)
+- `Most Recent`: served from etcd via quorum read
+- `Not older than`: at least as new as the provided `resourceVersion`
+- `Exact`: at the exact resource version provided
+- `Continue Token, Exact`: return data at the resource version of the initial paginated list call  
+- `Any`: strong consistency is not required, data at any resource version may be served
+
+We want to feed the semantics in to the width estimation process, we can use the following back of the
+envelope mapping:
+```
+S = {'Most Recent'=4 | 'Not older than'=2 | Exact=2 | 'Continue Token, Exact'=2 | 'Any'=1} 
```

Comment:
This is making it a magic from the very beginning and I think I have hard time with intuitive explanation of those constants. In particular, I see no reason why "Most Recent" and "Exact" should really have the same cost - they do exactly the same thing.

I would suggest dropping the S for now and if really needed leaving it for further releases.